pkg/worker/context.go

```diff
@@ -295,6 +302,7 @@ func (h *hatchetContext) saveOrLoadListener() (*client.WorkflowRunsListener, err
 		return h.listener, nil
 	}
 
+	// TODO: SHOULD THIS BE THE STEP CONTEXT? PROBABLY NOT
```

Comment:
@grutt flagging this for the Go SDK -- the shared listener is currently using the context from a single task, which it should definitely not be doing. it should likely be using the worker context (I'm not sure if this exists or not)